unstructured/ingest/connector/chroma.py

```diff
@@ -60,6 +62,18 @@ def initialize(self):
     def check_connection(self):
         _ = self.chroma_collection
 
+    def to_dict(self, **kwargs):
+        """
+        The _collection variable in this dataclass breaks deepcopy due to:
+        TypeError: cannot pickle '_thread.lock' object
+        When serializing, remove it, meaning client data will need to be reinitialized
+        when deserialized
+        """
+        self_cp = copy.copy(self)
+        if hasattr(self_cp, "_collection"):
+            setattr(self_cp, "_collection", None)
```

Comment:
Since we have it in dataclass initialization as a `None` value, is `setattr` to `None` better?
https://github.com/Unstructured-IO/unstructured/blob/4a34765fdfc09ec47ebe84dbacf50e6257afa65e/unstructured/ingest/connector/chroma.py#L44-L48